test/rules/ipg.js

```diff
@@ -0,0 +1,256 @@
+                subsectionContents: {definition: {title: "Definition", text: "The content of the definition"}}
+            };
+            formatted = ipg.formatSectionEntry(sectionEntry);
+        });
+
+        it("should contain the section's title", function() {
+            expect(formatted).to.contain(sectionEntry.title);
+        });
+        it("should contain the penalty, if there is one", function() {
+            expect(formatted).to.contain(sectionEntry.penalty);
+        });
+        it("should contain the section's text if present", function() {
+            expect(formatted).to.contain(sectionEntry.text);
+        });
+        it("should contain a preview if there is no text directly in the section", function () {
+            delete sectionEntry.text;
+            formatted = ipg.formatSectionEntry(sectionEntry);
+            const definitionEntry = sectionEntry.subsectionContents.definition;
+            expect(formatted).to.contain(definitionEntry.title).and.to.contain(definitionEntry.text);
+        });
+        it("should contain a list of available subsections", function () {
+            const availableSubsections = sectionEntry.subsections.map(s => `*${s}*`).join(', ');
+            expect(formatted).to.contain(availableSubsections);
+        });
+    });
+    describe("#formatSubsectionEntry", function() {
+        let sectionEntry;
+        let subsectionEntry;
+        let formatted;
+        let ipg;
+
+        beforeEach(function () {
+            ipg = new IPG();
+            sectionEntry = {
+                title: "1.1 A Section",
+                text: "Text contained in the section",
+                subsections: ["definition", "philosophy", "additional-remedy"],
+                subsectionContents: {definition: {title: "Definition", text: ["The content of the definition"]}}
+            };
+            subsectionEntry = sectionEntry.subsectionContents.definition;
+            formatted = ipg.formatSubsectionEntry(sectionEntry, subsectionEntry);
+        });
+
+        it("should contain the section's title and the subsection's title", function() {
+            expect(formatted).to.contain(sectionEntry.title).and.to.contain(subsectionEntry.title);
+        });
+        it("should contain the subsection's text if present", function() {
+            expect(formatted).to.contain(subsectionEntry.text);
+        });
+
+        it("should contain a list of other available subsections", function () {
+            const otherSubsections = sectionEntry.subsections.filter(s => s !== 'definition').map(s => `*${s}*`).join(', ');
+            expect(formatted).to.contain(otherSubsections);
+        });
+    });
+
+    describe("tests based on real data", function() {
+        let ipg;
+        this.timeout(10000);
+
+        before(function () {
+            ipg = new IPG();
+            return ipg.init();
+        });
+        it("should have downloaded and parsed the ipg", function () {
```

Comment:
It feels like once we go there, we could just stop downloading it altogether and always use an integrated version. As long as the bot breaks if the online document is a missing I don't really mind the tests breaking as well.

I can see the argument about running the tests without an internet connection though.